rust/src/pgsql/pgsql.rs

```diff
@@ -470,11 +473,16 @@ impl PgsqlState {
         while !start.is_empty() {
             match PgsqlState::state_based_resp_parsing(self.state_progress, start) {
                 Ok((rem, response)) => {
+                    if let Some(flow) = flow {
```

Comment:
I don't like that this is a runtime check only to make the tests work. Is there a cleaner solution?